### ISSUE TITLE
Handle connection errors

### ISSUE THREAD

**Issue by vikramsg** [FahrplanDatenGarten/pyhafas#29]
Hi,

This is a very nice project which I had been trying to use for my project. However, occasionally I hit the following error. 

```
 ('Connection aborted.', ConnectionResetError(54, 'Connection reset by peer'))
```

Usually, something like this can be handled by backoff, for example, in the `Request` class from `urllib3`. Would including this be interesting to you? Happy to contribute towards this.

**Comment by leona-ya** [comment]
Hey,
I currently don't have the time to implement this, but am happy to accept a PR :)
I'm not 100% sure how a good way of implementing would be, maybe a variable with a default on the profile class(es).

Just out of interest: With which profile did the error occur?

**Comment by vikramsg** [comment]
> With which profile did the error occur?

This is with the DBProfile.

> I'm not 100% sure how a good way of implementing would be, maybe a variable with a default on the profile class(es).

I haven't looked at detail but I would think an optional parameter with `retries` and `backoff_factor` would need to be set somewhere and all `requests` calls should use this.

**Comment by vikramsg** [comment]
Added a PR for this. 

https://github.com/FahrplanDatenGarten/pyhafas/pull/31

**Comment by leona-ya** [comment]
Now solved in #31, thanks to you!
